sleap_io/io/dataset.py

```diff
@@ -0,0 +1,270 @@
+from attr import Factory, asdict
+import h5py
+import numpy as np
+import pandas as pd
+import json
+from typing import List, Optional
+from sleap_io import (
+    Video,
+    Skeleton,
+    Edge,
+    Node,
+    Instance,
+    LabeledFrame,
+    Track,
+    Point,
+    PredictedInstance,
+)
+
+
+def from_pointsarray(
+    points: np.ndarray, skeleton: Skeleton, track: Optional[Track] = None
+) -> Instance:
+    """Create an `Instance` from an array of points.
+
+    Args:
+        points: A numpy array of shape `(n_nodes, 2)` and dtype `float32` that
+            contains the points in (x, y) coordinates of each node. Missing nodes
+            should be represented as `NaN`.
+        skeleton: A `sleap.Skeleton` instance with `n_nodes` nodes to associate with
+            the instance.
+        track: Optional `sleap.Track` object to associate with the instance.
+
+    Returns:
+        A new `Instance` object.
+    """
+    predicted_points = dict()
+    node_names: List[str] = [node.name for node in skeleton.nodes]
+    # TODO(LM): Ensure ordering of nodes and points match up.
+    for point, node_name in zip(points, node_names):
+        if (len(point)) == 4:
+            predicted_points[node_name] = Point(
+                x=point[0],
+                y=point[1],
+                visible=bool(point[2]),
+                complete=bool(point[3]),
+            )
+        else:
+            predicted_points[node_name] = Point(x=point[0], y=point[1])
+
+    return Instance(points=predicted_points, skeleton=skeleton, track=track)
+
+
+def from_instance(
```

Comment:
Rename function to describe `PredictedInstance` creation, i.e. `predicted_from_instance`